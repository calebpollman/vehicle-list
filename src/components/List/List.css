.list-container {
  align-items: center;
  background-color: var(--color-list-background);
  display: flex;
  flex-direction: column;
  height: var(--list-container-height);
  justify-content: center;
  overflow: scroll;
}