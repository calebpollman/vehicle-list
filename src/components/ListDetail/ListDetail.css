.list-detail-image {
  width: 100%;
}

.list-detail-title {
  color: var(--color-list-item-text);
  font-family: var(--font-application);
  font-size: var(--font-size-medium);
  font-weight: var(--font-weight-bold);
  padding: 2%;
}

.list-detail-text {
  color: var(--color-list-item-sub-text);
  font-family: var(--font-application);
  font-size: var(--font-size-small);
  padding-left: 2%;
}