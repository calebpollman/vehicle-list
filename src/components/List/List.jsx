import React from 'react';
import PropTypes from 'prop-types';
import ListItem from '../ListItem/ListItem';
import {LoadIcon} from '../Icons/Icons';
import './List.css';

const List = ({listLoading, searchActive, searchResults, vehicles}) => {
  
  const renderVehicles = (vehicles) => {
    return vehicles.map(vehicle => {
      const vehicleTitle = `${vehicle.year} ${vehicle.make} ${vehicle.model}`;
      return (
        <ListItem 
          key={vehicle.id}
          vehicle={vehicle}
          vehicleTitle={vehicleTitle}
        />
      );
    });
  }
  
  listLoading ? 
  vehicles = <LoadIcon /> : searchActive ? 
  vehicles = renderVehicles(searchResults) : vehicles = renderVehicles(vehicles);
  
  return (
    <ul className="list-container">
      {vehicles}
    </ul>
  );
}

List.propTypes = {
  listLoading: PropTypes.bool.isRequired,
  searchActive: PropTypes.bool.isRequired,
  searchResults: PropTypes.array.isRequired,
  vehicles: PropTypes.array.isRequired,
}

export default List;
