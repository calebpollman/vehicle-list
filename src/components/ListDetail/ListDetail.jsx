import React from 'react';
import PropTypes from 'prop-types';
import './ListDetail.css';

const ListDetail = ({vehicle}) => {
  const vehicleTitle = `${vehicle.year} ${vehicle.make} ${vehicle.model}`
  return (
    <div className="list-detail-container">
      <img className="list-detail-image" src={vehicle.image_url} alt="not found" />
      <p className="list-detail-title">{vehicleTitle}</p>
      <p className="list-detail-text">Mileage: {vehicle.mileage}</p>
    </div>
  );
}

ListDetail.propTypes = {
  vehicle: PropTypes.object.isRequired,
}

export default ListDetail;