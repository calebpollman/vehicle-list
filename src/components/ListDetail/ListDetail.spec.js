import React from 'react';
import {shallow} from 'enzyme';
import ListDetail from './ListDetail';
import List from '../List/List';

describe('ListDetail', () => {
  let wrapper;
  const vehicle = {
    "id": 0,
    "year" : "2013",
    "make" : "Kia",
    "model" : "Optima",
    "mileage" : 24235,
    "drivetrain" : "FWD",
    "bodytype" : "sedan",
    "image_url" : "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
    "created_at" : "2016-10-14T20:13:22.586Z"
  }

  beforeEach(() => {
    wrapper = shallow(<ListDetail vehicle={vehicle} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a wrapper <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });
});