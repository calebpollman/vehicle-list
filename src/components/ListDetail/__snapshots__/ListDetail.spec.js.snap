// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListDetail should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListDetail
    vehicle={
        Object {
            "bodytype": "sedan",
            "created_at": "2016-10-14T20:13:22.586Z",
            "drivetrain": "FWD",
            "id": 0,
            "image_url": "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
            "make": "Kia",
            "mileage": 24235,
            "model": "Optima",
            "year": "2013",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="not found"
          className="list-detail-image"
          src="http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg"
/>,
        <p
          className="list-detail-title"
>
          2013 Kia Optima
</p>,
        <p
          className="list-detail-text"
>
          Mileage: 
          24235
</p>,
      ],
      "className": "list-detail-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "not found",
          "className": "list-detail-image",
          "src": "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2013 Kia Optima",
          "className": "list-detail-title",
        },
        "ref": null,
        "rendered": "2013 Kia Optima",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Mileage: ",
            24235,
          ],
          "className": "list-detail-text",
        },
        "ref": null,
        "rendered": Array [
          "Mileage: ",
          24235,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="not found"
            className="list-detail-image"
            src="http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg"
/>,
          <p
            className="list-detail-title"
>
            2013 Kia Optima
</p>,
          <p
            className="list-detail-text"
>
            Mileage: 
            24235
</p>,
        ],
        "className": "list-detail-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "not found",
            "className": "list-detail-image",
            "src": "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2013 Kia Optima",
            "className": "list-detail-title",
          },
          "ref": null,
          "rendered": "2013 Kia Optima",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Mileage: ",
              24235,
            ],
            "className": "list-detail-text",
          },
          "ref": null,
          "rendered": Array [
            "Mileage: ",
            24235,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
